lib/rubygems/package.rb

```diff
@@ -711,18 +706,6 @@ def verify_files(gem)
     end
   end
 
-  ##
-  # Verifies that +entry+ is a valid gzipped file.
-
-  def verify_gz(entry) # :nodoc:
-    Zlib::GzipReader.wrap entry do |gzio|
-      # TODO: read into a buffer once zlib supports it
-      gzio.read 16_384 until gzio.eof? # gzip checksum verification
```

Comment:
It seems that we read until EOF because that's when the GZipReader will verify the gzip checksum header against the decompressed data. I read that from the doc https://docs.ruby-lang.org/en/3.4/Zlib/GzipReader.html

> When an reading request is received beyond the end of file (the end of compressed data). That is, when [Zlib::GzipReader#read](https://docs.ruby-lang.org/en/3.4/Zlib/GzipReader.html#method-i-read),

Now that we are no longer reading anywhere the whole gzip since I think we early return in various places e.g. here https://github.com/ruby/rubygems/blob/1a30a709b1ded46df6d18a5083be9f9e718ed9ea/lib/rubygems/package.rb#L395-L399

Is it ok to not have this checksum verification anymore and verify the integrity of the zip ?